Adds the option to try cloaked search with open search as well as elastic search (#15)
IronCoreLabs/try-cloaked-search: README.md

### (optional) Look at an encrypted index

> _Note that all queries made with `./query-search-service.sh` are being made directly to your search service. We are using a script to detect if you're running open search or elastic search, but there is no involvement from cloaked search. Since these requests go directly to the search service (port 9200) so we can see what's actually stored._

Reviewer: ```suggestion
_Note that all queries made with `./query-search-service.sh` are being made directly to your search service. We are using a script to detect if you're running OpenSearch or Elasticsearch, but there is no involvement from Cloaked Search. Since these requests go directly to the search service (port 9200), we can see what's actually stored._
```

Let's get all the documents belonging to `tenant-1` and see what's in the index!

```bash

> ./query-search-service.sh "+tenant_id:"tenant-1""

Reviewer: ```suggestion
./query-search-service.sh '+tenant_id:"tenant-1"'
```

The data not associated with any tenant is readable in the clear. If we do a very generic query, some documents with no tenant will come back.

```bash

> ./query-search-service.sh "title:list"

Reviewer: ```suggestion
./query-search-service.sh 'title:list'
```

These are a couple examples of simple term queries. They are still querying on the `title` field, which is unprotected.

```bash
./query-cloaked-search.sh '+tenant_id:"tenant-1" AND title:Japan'
./query-cloaked-search.sh '+tenant_id:"tenant-1" AND title:cup'
```

> Compare these results to the ones returned by querying Elasticsearch directly. The same documents are returned, but the contents are much different.

Reviewer: I know this wasn't changed in this PR :)
```suggestion
Compare these results to the ones returned by querying Elasticsearch directly. The same documents are returned, but the contents are very different.
```

You can see how Cloaked Search transparently handles the decryption of the document to allow you to see the data in the fields that were protected, `summary` and `body`.

```bash
./query-search-service.sh '+tenant_id:"tenant-1" AND title:Japan'
./query-search-service.sh '+tenant_id:"tenant-1" AND title:cup'
```

Now try querying on a protected field:

```bash
./query-cloaked-search.sh '+tenant_id:"tenant-1" AND summary:glasgow'
```

Term queries can also be combined with ORs or ANDs, and you can mix protected and unprotected fields. For example,

```bash
./query-cloaked-search.sh '+tenant_id:"tenant-1" AND (title:cup OR title:Japan)'
./query-cloaked-search.sh '+tenant_id:"tenant-1" AND (summary:cup OR title:Japan)'
./query-cloaked-search.sh '+tenant_id:"tenant-1" AND (summary:cup OR body:Japan)'
```

Phrases can also be searched using quoted queries like this:

```bash
./query-cloaked-search.sh '+tenant_id:"tenant-1" AND summary:"Cheerleading in Japan"'
```

Finally, here is an example of a prefix query:

```bash

> ./query-cloaked-search.sh '+tenant_id:"tenant-1" AND summary:list*'

Reviewer: I know this is a query from before, but it's not a very good prefix query as all the results are just for "List", so the substring isn't adding any visible benefit
